File: day1/day1.cpp
#include "/Users/seelengxd/stdc++.h"
 
typedef long long ll;
typedef pair<ll, ll> pll;
#define all(a) a.begin(),a.end()
#define pb push_back
#define mp make_pair
#define forn(i,n) for(int i = 0; i < int(n); ++i)
#define forkn(i,k,n) for(int i = k; i < int(n); ++i)
#define sz(a) int(a.size())
#define deb(x) cout << #x << " " << x << endl;
 
template<typename... T>
void read(T&... args){
    ((cin >> args), ...);
}
 
template<typename... T>
void write(T&&... args){
    ((cout << args << " "), ...);
}
 
mt19937 rng(chrono::steady_clock::now().time_since_epoch().count());

void skipline() {
    cin.ignore(numeric_limits<streamsize>::max(), '\n');
}

void update(priority_queue<int, vector<int>, greater<int>>* pq, int curr) {
    pq->push(curr);
    if (pq->size() > 3) {
        pq->pop();
    }
}
 
void solve() {
    int curr = 0;
    priority_queue<int, vector<int>, greater<int>> pq;
    forn(i, 2) skipline();
    string line;
    while (getline(cin, line)) {
        if (line.empty()) {
            update(&pq, curr);
            curr = 0;
        } else {
            curr += stoi(line);
        }
    }
    update(&pq, curr);
    int third = pq.top();
    pq.pop();
    int second = pq.top();
    pq.pop();
    int first = pq.top();
    cout << "Part 1: " << first << endl;
    cout << "Part 2: " << first + second + third << endl;
    
}
 
int main() {
    ios::sync_with_stdio(false);
    solve(); 
}

File: day1/day1.ts
const fs = require('fs');

const data : string = fs.readFileSync('day1.txt', 'utf8');

function day1() {
    let first = 0;
    let second = 0;
    let third = 0;
    let curr = 0;

    function updateTopThree(curr: number) : void {
        if (first < curr) {
            third = second;
            second = first;
            first = curr;
        } else if (second < curr) {
            third = second;
            second = curr;
        } else if (third < curr) {
            third = curr;
        }
    }

    for (const line of data.split("\n")) {
        if (!line) {
            updateTopThree(curr);
            curr = 0;
        } else {
            curr += parseInt(line);
        }
    }
    
    updateTopThree(curr);

    console.log(`Part 1: ${first}`)
    console.log(`Part 2: ${first + second + third}`);
}

day1();
